pandas/core/arrays/numpy_.py

```diff
@@ -147,7 +148,11 @@ def _from_sequence(cls, scalars, dtype=None, copy=False):
         if isinstance(dtype, PandasDtype):
             dtype = dtype._dtype
 
-        result = np.asarray(scalars, dtype=dtype)
+        with warnings.catch_warnings():
+            # See https://github.com/numpy/numpy/issues/15041
+            warnings.filterwarnings("ignore", ".*with automatic object dtype.*")
+            result = np.asarray(scalars, dtype=dtype)
```

Comment:
This should be fixed at the caller, not here

Reply:
yah, pretty much every usage where we're just suppressing the warnings was a kludge

Reply:
My thought for a way forward with the new sentinel in numpy/numpy#15119 is that you would leave this unchanged. When needed you would add the sentinel as the dtype in the call to `_from_sequence` (or even to whoever is calling `_from_sequence`. During the deprecation period, try to rework the use case that justifies the inefficient ragged array, and get back to NumPy with the use case for such a construct so we can stop or rethink the deprecation.